add_address in main.py

Opetry16 opened the pull request:
1. Defined add_address function
2. Add new command "add-address" in COMANDS
3. New command included in HELP

Opetry16 committed:
add_address in main.py

1. Defined add_address function
2. Add new command "add-address" in COMANDS
3. New command included in HELP

kryvosheyin commented:
@Opetry16 додай будь ласка ще клас Address який буде наслідуватися від поля Field з методом ініціалізаціі ( __init__ ) який також буде мати валідацію поля 

Також треба додати функцію зберігання адреси в самому класі Record. бо зараз якщо ти викличеш цей метод з main, то нічого не відбудеться. покаже месседж, що адреса додалася, але по факту її не буде

Opetry16 committed:
Create Address Class

Opetry16 committed:
Rename Address to Address.py

Opetry16 committed:
Update Record.py

Now, the Record class includes an add_address method to set the address for a contact, and it also includes the address in the __str__ method to display the address when you print a Record object.

Opetry16 commented:
@kryvosheyin подивись будь ласка, добавив те що ти порадив. дякую

kryvosheyin commented on src/classes/Record.py:
```diff
@@ -52,4 +56,5 @@ def get_phones(self):
         birthday_str = str(self.birthday) if self.birthday else "Not set"
-        return f"Contact name: {self.name.value}, phones: {';' .join(p.value for p in self.phones)}, birthday: {birthday_str}"
+        address_str = str(self.address) if self.address else "Not set"
+        return f"Contact name: {self.name.value}, phones: {';' .join(p.value for p in self.phones)}, birthday: {birthday_str}", address: {address_str}"
```
тут валідатор викине помлку, тому що лапки закрилися перед address: {address_str}. щоб код працював, їх треба прибрати

kryvosheyin commented on src/classes/Address.py:
```diff
@@ -0,0 +1,11 @@
+
+    def validate(self):
+        if not self.value:
+            raise IncorrectAddressFormatException("Address cannot be empty")
```
тут здається доданий новий кастомний ексепшн IncorrectAddressFormatException, але його також треба додати в клас Exceptions.py та імпортувати у цей файл, щоб працювало.

kryvosheyin commented on src/classes/Address.py:
```diff
@@ -0,0 +1,11 @@
+class Address(Field):
```
Класс Address наслідується від класу Field, але його треба імпортувати, щоб не виникло помилки. на даному етапі якщо запустиш програму, то вона не запуститься

kryvosheyin reviewed (changes requested):
@Opetry16 будь ласка подивись коменти та виправ. дякую

kryvosheyin commented on src/main.py:
```diff
@@ -140,6 +140,13 @@ def show_birthday(address_book: AddressBook, args):
 
+@input_error
+def add_address(address_book: AddressBook, args):
+    name, address = extract_args(args)
```
@Opetry16 змінна address в даному випадку приймає лише один аргумент який йде після name. 
тобто, якщо запустити функцію так `add-address alex some address` - то вона не спрацює, бо в змінну буде передаватися тільки `some`, а все що далі - то будуть наступні аргументи. 

Твоя задача вирішити як зробити так, щоб все що йшло після імені контакту зчитувалося як рядок та записувалося у змінну address. 

для прикладу виклик такої команди `add-address Oleg 123 South Drive, New York, NY, 10023` повинен знайти юзера Oleg у книзі контактів та додати туди повну адресу 123 South Drive, New York, NY, 10023

Opetry16 committed:
Update Record.py

забрав лапки перед {address_str}

Opetry16 committed:
Update Exceptions.py

 доданий новий кастомний ексепшн IncorrectAddressFormatException, в клас Exceptions.py

Opetry16 committed:
Update Address.py

імпортував  IncorrectAddressFormatException,

Opetry16 committed:
Update Address.py

додав from classes.Fields import Field

Opetry16 committed:
Update main.py

зараз все що йде після імені контакту зчитується як рядок та записується у змінну address.
